astropy/coordinates/sites.py

```diff
@@ -0,0 +1,203 @@
+Longitudes are listed with positive to the West.
+
+Additions or corrections to the observatory list can be submitted via Pull
+Request to the [astropy GitHub repository](https://github.com/astropy/astropy),
+updating the ``observatories.json`` file.
+
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import json
+from difflib import get_close_matches
+
+from ..utils.data import get_pkg_data_contents
+from ..extern import six
+from .earth import EarthLocation
+
+__all__ = ['get_site', 'get_site_names', 'add_site', 'remove_site']
+
+# Observatory database and list of names:
+_site_db = None
+_site_names = []
+
+def _load_sites():
+    """
+    Load observatory database from data/observatories.json and parse them into
+    a dictionary in memory.
+    """
+    global _site_db, _site_names
+    _site_db = dict()
+    db = json.loads(get_pkg_data_contents('data/observatories.json'))
+    for site in db:
+        location = EarthLocation.from_geodetic(db[site]['longitude'],
+                                   db[site]['latitude'],
+                                   db[site]['elevation_meters'])
+        _site_names.append(db[site]['name'])
+        for alias in db[site]['aliases']:
+            _site_db[alias.lower()] = location
+
+def get_site(site_name):
+    """
+    Return an `~astropy.coordinates.EarthLocation` object for known observatory.
+
+    Use `~astropy.coordinates.get_site_names` to see a list of available
+    observatories.
+
+    Parameters
+    ----------
+    site_name : str
+        Name of the observatory.
+
+    Returns
+    -------
+    `~astropy.coordinates.EarthLocation`
+        The location of the observatory.
+
+    See Also
+    --------
+    get_site_names : the list of sites that this function can access
+    add_site : adds a new site to the list
+    remove_site : remove a site by name from the list of sites
+    """
+    if _site_db is None:
+        _load_sites()
+
+    if site_name.lower() not in _site_db.keys():
+        # If site name not found, find close matches and suggest them in error
+        close_names = get_close_matches(site_name, _site_db.keys())
+        close_names = sorted(close_names, key=lambda x: len(x))
+        if len(close_names) > 0:
+            errmsg = ("Site not in database. Use ``get_site_names()`` "
+                      "to see available sites. Did you mean: '{}'?".format(
+                      "', '".join(close_names)))
+        else:
+            errmsg = 'Site not in database.'
+        raise KeyError(errmsg)
+
+    return _site_db[site_name.lower()]
+
+def get_site_names(full_list=True):
+    """
+    Get list of names of observatories for use with
+    `~astropy.coordinates.get_site`.
+
+    Parameters
+    ----------
+    full_list : bool
+        Show full list observatory names and aliases (True), or just the list
+        of names (False)? Default to True.
```

Comment:
I think should be named something more explicit like `show_aliases`.  `full_list` makes me think "Is it not showing me all the observatories by default for some reason?"